### PR TITLE
fix(logging): redact the home directory on Windows too

### PR DESCRIPTION
**Pull request by Virgile-pct**
On Windows the logging redaction never hid the home directory, because it looked the home up in an environment variable that Windows does not set.

## What happens today

```rust
pub fn sanitize_paths_in_text(text: &str) -> String {
    use std::env;
    if let Some(home) = env::var_os("HOME") {
        ...
    } else {
        text.to_string()          // Windows lands here, every time
    }
}
```

Measured on Windows 11: a plain process sees `USERPROFILE`, `HOMEDRIVE` and `HOMEPATH`, and no `HOME`. So both `sanitize_path` and `sanitize_paths_in_text` returned their input untouched.

This is not confined to tests. `sanitize_paths_in_text` is called from:

- `logging/persistence.rs:343`, on persisted detail JSON
- `logging/policy/artifact_redaction.rs:50` and `:93`
- `logging/service/operational_audit/context.rs:236`

So persisted details, redacted artifacts and operational audit context all kept `C:\Users\<name>` in clear text on Windows. Given how much care the log store already gets there, with its own ACL checks in the platform slice, this looked worth fixing rather than reporting.

## What this changes

Both functions resolve the home through `dirs::home_dir()`, which is what the rest of the crate already does, including `logging/foundation.rs` in the same module. On Unix `dirs` reads `$HOME` first, so behaviour there is unchanged.

## The tests were POSIX-shaped too

With the home resolving, the two tests still failed. Probing the real output showed why:

```
home    = C:\Users\<me>
input   = C:\Users\<me>\some/deep/path/file.log
output  = ~\some/deep/path/file.log
contains the home = false
```

The redaction was already correct; the assertions wanted a `~/` prefix, and a redacted Windows path reads `~\`. They now build the expected prefix from `std::path::MAIN_SEPARATOR`, so they check the same shape on both platforms.

`sanitize_paths_in_text_replaces_home` only asserted that a tilde appeared, and never that the home was gone. It now asserts both, which is the property the function exists to provide.

## Measured effect

Full suite on Windows 11, against `main` at `3f4f1c35a`:

| | `main` | this branch |
|---|---|---|
| passed | 3457 | 3459 |
| failed | 17 | 15 |

The two redaction tests are the whole difference. No new failures, and the other 15 are identical, compared by diffing both complete sorted lists.

## One thing I did not do

The output keeps the platform separator, so a redacted Windows path reads `~\some\path`. Normalising every logged path to forward slashes would make artifacts compare cleanly across platforms, but it changes what Windows users see in their own logs, so it did not belong in a privacy fix. Say the word if you would rather have it.

## Gates

- `cargo test -p mesh-llm-host-runtime`: 3459 passed, 15 failed, as above.
- `cargo fmt --all --check`: clean.
- `cargo clippy -p mesh-llm-host-runtime --all-targets`: 8 lints, complete sorted set identical to `main`.
- `cargo run -p xtask -- repo-consistency no-console-print`: passed.
- `scripts/check-env-mutation-contract.py` and `python -m unittest scripts.tests.test_env_mutation_contract`: passed.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved privacy in audit logs across platforms, including systems without a standard home-directory environment variable.
  - Redacts exact home-directory paths as `~` while preserving platform-specific separators and leaving similarly named sibling paths unchanged.
  - Improved redaction in repeated paths, prose, punctuation, whitespace, URLs, and structured JSON, including nested objects and arrays.
  - Non-JSON content continues to receive standard text redaction.

- **Tests**
  - Added coverage for platform-specific paths, structured content, URL parameters, and edge cases.

### CODE REVIEW

File: crates/mesh-llm-host-runtime/src/logging/policy.rs [hunk 1]
```diff
@@ -451,10 +451,14 @@ fn redact_key_in_json(text: &str, key: &str) -> String {
...
+///
+/// The home directory comes from `dirs::home_dir()`, as everywhere else in this
+/// crate. Reading `HOME` directly redacted nothing on Windows, where that
+/// variable is not set and the home lives behind the known-folder API.
 pub fn sanitize_path(path: &std::path::Path) -> String {
-    use std::env;
-    if let Some(home) = env::var_os("HOME") {
+    if let Some(home) = dirs::home_dir() {
         let home_str = home.to_string_lossy();
         return path.to_string_lossy().replace(&*home_str, "~");
```
[the remark this answers is not in the document]
Comment (Virgile-pct) [reply crates/mesh-llm-host-runtime/src/logging/policy.rs]: Good catch, and reproduced before acting on it.

With a home of `C:\Users\<me>`, a sibling directory came out mangled exactly as you describe:

```
home    = C:\Users\<me>
sibling = C:\Users\<me>-backup\log.txt
output  = ~-backup\log.txt
```

The flaw predates this branch, but you are right that resolving the home is what makes it reachable on Windows, where the lookup previously returned early on an unset `HOME`.

Fixed in 0465f246e, following both of your suggestions. `sanitize_path` takes a `Path`, so it now uses `strip_prefix` and compares components rather than characters; a sibling fails to strip and is returned unchanged. `sanitize_paths_in_text` has only free-form text to work with, so it replaces a match only when a separator or the end of the input follows.

`strip_prefix` also surfaced an edge neither of us mentioned: the home on its own yields an empty remainder, which would have left a trailing separator. It now collapses to `~`.

Two tests cover both cases.

**Review by michaelneale — DISMISSED** [review]
Jimmy (bot review): The Windows home lookup fix is good, but the added boundary check regresses macOS/Linux redaction: `{"home":"/Users/alice"}` now retains the home path because a closing quote is not accepted as a boundary. Please keep the lookup fix and preserve the previous text-redaction coverage, with a regression test.

**Review by michaelneale — DISMISSED** [review]
thanks @Virgile-pct just tidying it up and will get it in

File: crates/mesh-llm-host-runtime/src/logging/policy.rs [hunk 2]
```diff
@@ -481,9 +492,10 @@ pub fn hash_value(input: &str) -> String {
...
 
 /// Remove private directory prefixes from a string containing paths.
+/// Unlike `sanitize_path`, conservatively redact every occurrence of the home
+/// string, including bare homes in prose and prefixes shared with siblings.
 pub fn sanitize_paths_in_text(text: &str) -> String {
-    use std::env;
-    if let Some(home) = env::var_os("HOME") {
+    if let Some(home) = dirs::home_dir() {
         let home_str = home.to_string_lossy().to_string();
         text.replace(&home_str, "~")
```
[the remark this answers is not in the document]
Comment (michaelneale) [reply crates/mesh-llm-host-runtime/src/logging/policy.rs]: Fixed in a383d19118b3ab5a90eb04b2f75fd2fbbdb9cf6a. Audit persistence now calls a JSON-specific sanitizer which parses the serialized detail and walks decoded string values (including nested arrays/objects), then serializes them again. The free-form sanitizer remains boundary-aware without acquiring JSON parsing semantics. Replaced the fake JSON fixture with serde_json serialization and added deterministic backslash-Windows, forward-slash-Windows, Unix, quoted-home, descendant, repeated/bare-home and sibling-preservation cases. Ten path-module contracts passed in a small standalone harness importing the actual source; warning-denying harness Clippy and workspace formatting passed. Full package/platform validation is left to CI as requested by the maintainer.

**Comment by michaelneale** [comment]
@coderabbitai review

File: crates/mesh-llm-host-runtime/src/logging/policy/path_redaction.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+}
+
+fn is_text_delimiter(c: char) -> bool {
+    // Whitespace and these punctuation marks are interpreted as prose, even
+    // though some filesystems permit them within a component.
+    c.is_whitespace()
+        || matches!(
+            c,
+            '\"' | '\'' | '`' | ')' | ']' | '}' | ',' | ';' | ':' | '!' | '?'
```
[the remark this answers is not in the document]
Comment (michaelneale) [reply crates/mesh-llm-host-runtime/src/logging/policy/path_redaction.rs]: Verified this boundary omission and fixed in b114fedf6c9780ae8544ee209a15ac6fe7011ed5. Added & to the text delimiters and deterministic Unix/Windows URL fixtures for both & and &amp;. All 11 isolated source-module contracts and warning-denying harness Clippy pass; workspace formatting passes. JSON persistence correction remains in a383d19118b3ab5a90eb04b2f75fd2fbbdb9cf6a.
